pkg/cmd/synonyms/save/save.go

```diff
@@ -0,0 +1,110 @@
+package save
+
+import (
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/spf13/cobra"
+
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/opt"
+	"github.com/algolia/cli/pkg/cmdutil"
+	validator "github.com/algolia/cli/pkg/cmdutil/validators"
+	"github.com/algolia/cli/pkg/cmdutil/wording"
+)
+
+// NewSaveCmd creates and returns a save command for index synonyms
+func NewSaveCmd(f *cmdutil.Factory, runF func(*validator.SaveOptions) error) *cobra.Command {
+	opts := &validator.SaveOptions{
+		IO:           f.IOStreams,
+		Config:       f.Config,
+		SearchClient: f.SearchClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:               "save <index> --id <id> --synonyms <synonyms>",
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: cmdutil.IndexNames(opts.SearchClient),
+		Short:             "Save a synonym to the given index",
+		Aliases:           []string{"create", "edit"},
+		Long: heredoc.Doc(`
+			This command save a synonym to the specified index.
+			If the synonym doesn't exist yet, a new one is created.
+		`),
+		Example: heredoc.Doc(`
+			# Save one standard synonym with ID "1" and "foo" and "bar" synonyms to the "TEST_PRODUCTS_1" index
+			$ algolia save TEST_PRODUCTS_1 --id 1 --synonyms foo,bar
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.Indice = args[0]
+
+			synonym, err := validator.ValidateFlags(*opts)
```

Comment:
flags validation and search.Synonym creation depending of the type here: the logic is hidden inside `ValidateFlags(...)`